foxtrot-core/src/main/java/com/flipkart/foxtrot/core/common/Action.java

```diff
@@ -71,7 +71,10 @@ public String cacheKey() {
         return cacheKey;
     }
 
-    public AsyncDataToken execute(ExecutorService executor) {
+    public AsyncDataToken execute(ExecutorService executor) throws QueryStoreException {
+        if (!validate()) {
+            throw new QueryStoreException(QueryStoreException.ErrorCode.NO_SUCH_TABLE, "Table/s not found");
```

Comment:
Let validate throw exception instead of throwing same ErrorCode for all validation failures